### PR TITLE
LPS-148344 [CP 2.0] Add customer notification for key expiration when generating an activation key

### PR DESCRIPTION
**Pull request by eduardo-diniz**
[LPS-148344](https://issues.liferay.com/browse/LPS-148344)

### CODE REVIEW

**Comment by eduardo-diniz** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 1]
```diff
@@ -426,7 +453,34 @@ const RequiredInformation = ({
 										</Button>
 									</ClayTooltipProvider>
 
-									<div className="dropdown-divider"></div>
+									{featureFlag.includes('LPS-153478') && (
```
Comment (samuelleolopes) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:456]: Replace with FeatureFlags ['LPS-153478']

**Review by samuelleolopes — APPROVED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx (hunk 1 above)
Comment (eduardo-diniz) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:456]: Done! Thx

**Comment by samuelleolopes** [comment]
Good Job! :fire:

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 2]
```diff
@@ -166,16 +177,25 @@ const RequiredInformation = ({
 			startDate: infoSelectedKey?.selectedSubscription.startDate,
 		};
 
+		const subscriptionKey = async (id) => {
+			await putSubscriptionInKey(provisioningServerAPI, id, sessionId);
+		};
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:182]: Simplify

```suggestion
		const saveSubscriptionKey = (id) => {
			return putSubscriptionInKey(provisioningServerAPI, id, sessionId);
		};
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 3]
```diff
@@ -166,16 +177,25 @@ const RequiredInformation = ({
...
 				accountKey,
 				provisioningServerAPI,
 				sessionId,
 				licenseKey
 			);
-		}
-		else {
-			await Promise.all(
+			await subscriptionKey(results[0].items.id);
+			setIsLoadingUserInvitation(false);
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:194]: 1. Spaces
2. Is this right `results[0].items.id` ?  or `results.items[0].id` ?

```suggestion

			await saveSubscriptionKey(results[0].items.id);
			
			setIsLoadingUserInvitation(false);
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 4]
```diff
@@ -166,16 +177,25 @@ const RequiredInformation = ({
...
+			const results = await createNewGenerateKey(
 				accountKey,
 				provisioningServerAPI,
 				sessionId,
 				licenseKey
 			);
-		}
-		else {
-			await Promise.all(
+			await subscriptionKey(results[0].items.id);
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:193]:
```suggestion
			await saveSubscriptionKey(results[0].items.id);
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
Good job @eduardo-diniz , see

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx [hunk 5]
```diff
@@ -426,7 +453,34 @@ const RequiredInformation = ({
...
+											<div className="d-flex">
+												<div className="pr-2 pt-1">
+													<ClayCheckbox
+														checked={
+															checkedBoxSubscription
+														}
+														id="expiration-checkbox"
+														onChange={() =>
+															setCheckedBoxSubscription(
+																(val) => !val
```
[the remark this answers is not in the document]
Comment (eduardo-diniz) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:467]: thanks!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx (hunk 3 above)
Comment (eduardo-diniz) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.jsx:194]: The correct is this: results[0].items[0].id

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Review by kevenleone — APPROVED** [review]
LGTM 🔥

**Pull request closed without merging**